src/Object/Datasource/DatasourceProperties.php

```diff
@@ -4,30 +4,72 @@
 
 /**
  * Class DatasourceProperties
+ * @property mixed parameters
  * @package ExpandOnline\KlipfolioApi\Object\Datasource
  */
 class DatasourceProperties extends BaseApiResource
 {
 
-    const FIELD_PROPERTIES = 'properties';
+    const FIELD_PARAMETERS = 'parameters';
+
+    /**
+     * Extends base constructor so we can set $this->parameters to an instance of DatasourceProperties
+     * @param array $data
+     */
+    public function __construct($data = [])
+    {
+        parent::__construct($data);
+
+        if (!empty($this->data[static::FIELD_PARAMETERS])) {
+            $this->data[static::FIELD_PARAMETERS] = new DatasourcePropertiesParameters($this->data[static::FIELD_PARAMETERS]);
+        }
+    }
 
     /**
      * @return mixed
      */
-    public function getProperties() {
-        return $this->data[static::FIELD_PROPERTIES];
+    public function getProperties()
+    {
+        return array_diff_key($this->data, array_flip(['id']));
```

Comment:
Properties API response doesn't return ID, but BaseApiResource assumes we always have an ID. Thus it will just output `{id: NULL, actualProperty1: value1, actualProperty2: value2}` for every instance